Cargo.toml

```diff
@@ -44,3 +47,6 @@ pgx-tests = "0.3.1"
 pgx = { git = "https://github.com/timescale/pgx", branch = "promscale-staging" }
 pgx-macros = { git = "https://github.com/timescale/pgx", branch = "promscale-staging" }
 pgx-tests = { git = "https://github.com/timescale/pgx", branch = "promscale-staging" }
+
+# from e2e workspace
+test-generator = { git = "https://github.com/JamesGuthrie/test-generator" }
```

Comment:
Why do we need it at the top level? Can't it live in the e2e workspace?

Reply:
https://doc.rust-lang.org/cargo/reference/workspaces.html#key-features

> The key points of workspaces are:
> - All packages share a common Cargo.lock file which resides in the workspace root.
> - All packages share a common [output directory](https://doc.rust-lang.org/cargo/guide/build-cache.html), which defaults to a directory named target in the workspace root.
> - **The [[patch]](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section), [[replace]](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-replace-section) and [[profile.*]](https://doc.rust-lang.org/cargo/reference/profiles.html) sections in Cargo.toml are only recognized in the root manifest, and ignored in member crates' manifests.**